samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -176,6 +180,75 @@ private ApplicationConfig buildStreamApplicationConfig(String systemName, String
   }
 
   @Test
+  public void shouldStopNewProcessorsJoiningGroupWhenNumContainersIsGreaterThanNumTasks() throws InterruptedException {
+    /**
+     * sspGrouper is set to AllSspToSingleTaskGrouperFactory for this test case(All ssp's from input kafka topic are mapped to a single task).
+     * Run a stream application(streamApp1) consuming messages from input topic(effectively one container).
+     *
+     * In the callback triggered by streamApp1 after processing a message, bring up an another stream application(streamApp2).
+     *
+     * Assertions:
+     *           A) JobModel generated before and after the addition of streamApp2 should be equal.
+     *           B) Second stream application(streamApp2) should not join the group and process any message.
```

Comment:
Here's the JIRA for the issue we discussed offline: https://issues.apache.org/jira/browse/SAMZA-1364. 
I'll add a comment in the test.